src/NUnitFramework/framework/TestContext.cs

```diff
@@ -295,6 +295,16 @@ public static void AddFormatter(ValueFormatterFactory formatterFactory)
         }
 
         /// <summary>
+        /// Attach a file to the current test result
+        /// </summary>
+        /// <param name="filepath">Relative or absolute filepath to attachment</param>
+        /// <param name="description">Optional description of attachment</param>
+        public static void AddTestAttachment(string filepath, string description = null)
+        {
+            TestExecutionContext.CurrentContext.AddTestAttachment(filepath, description);
```

Comment:
I can change that if this ways preferred. I get a little mixed up about logic should live between TestContext and TestExecutionContext - I'd followed how `AddFormatter()` was written.

This is also more possible now I'm no longer storing an intermediate list of attachments on TestExecutionContext. 😄